### DISCUSSION TITLE
Ideal Timeout for large dataset 1000,1500,2000,2500 stores

### DISCUSSION THREAD

**Discussion by SruthyJohn** [google/or-tools#3221] [Routing (and legacy CP) questions]
Hi Team,
Can you please suggest an ideal time limit to obtain feasible solution for 1000, 1500, 2000, 2500 stores.
I have tested with 500 and it returned results; but while testing with different timelimits (ie, 2,3,5,10,15,20 minutes) and metaheuristic algorithm the results varies with different TotalTime, TotalDistance. 

<html>
<body>
<google-sheets-html-origin><style type="text/css"></style>

No. of Coordinates | No.of Vans | First Solution Strategy | Metaheuristic Algorithm | Time Out/Limit (minutes) | Time Taken | Total Distance for the Routes (kms) | Total Time for the Routes (Hours)
-- | -- | -- | -- | -- | -- | -- | --
500 | 30 (25 each) | PathCheapestArc | - | Nil | 2 | 672.25 | 13:35 hours
500 | 30 (25 each) | PathCheapestArc | Automatic | 5 | Nil | 672.25 | 13:35 hours
500 | 30 (25 each) | PathCheapestArc | TabuSearch | 2 | 2 | 665.434 | 13:28 hours
500 | 30 (25 each) | PathCheapestArc | TabuSearch | 3 | 3 | 665.434 | 13:28 hours
500 | 30 (25 each) | PathCheapestArc | TabuSearch | 4 | 4 | 665.434 | 13:28 hours
500 | 30 (25 each) | PathCheapestArc | TabuSearch | 5 | 5 | 665.434 | 13:28 hours
500 | 30 (25 each) | PathCheapestArc | TabuSearch | 10 | 10 | 665.434 | 13:28 hours
500 | 30 (25 each) | PathCheapestArc | TabuSearch | 15 | 15 | 665.434 | 13:28 hours
500 | 30 (25 each) | PathCheapestArc | TabuSearch | 20 | 20 | 665.434 | 13:28 hours
500 | 30 (25 each) | PathCheapestArc | GuidedLocal Search | 2 | 2 | 672.25 | 13:35 hours
500 | 30 (25 each) | PathCheapestArc | GuidedLocal Search | 3 | 3 | 669.13 | 13:32 hours
500 | 30 (25 each) | PathCheapestArc | GuidedLocal Search | 5 | 5 | 698.14 | 13:49 hours
500 | 30 (25 each) | PathCheapestArc | GuidedLocal Search | 10 | 10 | 682.95 | 13:29 hours
500 | 30 (25 each) | PathCheapestArc | GuidedLocal Search | 15 | 15 | 682.95 | 13:29 hours
500 | 30 (25 each) | PathCheapestArc | GuidedLocal Search | 20 | 20 | 682.95 | 13:29 hours

</body>
</html>

Can you please suggest an ideal TimeLimit for big data set?

**Comment by jmarca** [comment]
As far as I understand things, there is no ideal time limit.  The running time is highly dependent on the problem instance and your formulation.  What you are doing is the best way forward—try different times and record the results, and pick a time you are comfortable with. 

I have one problem domain/client where the inputs can be wildly different across program runs, which leads to highly variable “ideal” times. In that case, what I’ve done is to set up a callback function that evaluates the percent change in the objective over time. If enough time elapses with little to no change, then I bail out.

I have also used both numbers of solutions and time.  For example, 900 seconds and 900 solutions.  The easy problems blow through the solutions lickety split, so they finish quickly, and the hard problems are slower, and tend to time out rather than hitting the solution limit. It is easier to set up than the callback system, but I prefer the callback approach because I have better control over the outcome. 

Regards,

James
